include/exadg/grid/mapping_dof_vector.h

```diff
@@ -32,22 +32,89 @@
 #include <deal.II/fe/fe_system.h>
 #include <deal.II/fe/fe_tools.h>
 #include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/mapping_fe_field.h>
 #include <deal.II/fe/mapping_q_cache.h>
 #include <deal.II/lac/la_parallel_block_vector.h>
 #include <deal.II/multigrid/mg_transfer_matrix_free.h>
 
 // ExaDG
 #include <exadg/grid/grid_data.h>
+#include <exadg/operators/finite_element.h>
 #include <exadg/solvers_and_preconditioners/multigrid/transfer.h>
 
 namespace ExaDG
 {
+namespace MappingTools
+{
 /**
- * A mapping class based on dealii::MappingQCache equipped with practical interfaces that can be
- * used to initialize the mapping.
+ * Extract the grid coordinates for a given external mapping and fill a
+ * dof-vector given a corresponding dealii::DoFHandler object.
+ */
+template<int dim, typename Number>
+void
+fill_grid_coordinates_vector(dealii::Mapping<dim> const &                         mapping,
```

Comment:
@kronbichler similar to comment https://github.com/exadg/exadg/pull/634/files#r1457362322, I wonder whether we could realize this function as `dealii::VectorTools::interpolate()` using an "identity function" as `dealii:Function`.

Reply:
Do you mean `VectorTools::get_position_vector`, https://www.dealii.org/current/doxygen/deal.II/namespaceVectorTools.html#afcb40e50d2fa4807762faefed206f5e3

I did not look closely, but it might work, and we might possibly want to optimize the function there with the optimization we developed for this PR.

Reply:
Originally, I meant `VectorTools::interpolate()`, using a function `f(x) = x` (identity).